packages/http-client-csharp/generator/Microsoft.Generator.CSharp.Input/src/InputTypes/InputType.cs

```diff
@@ -1,18 +1,21 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System.Collections.Generic;
+using Microsoft.Generator.CSharp.Input.InputTypes;
+
 namespace Microsoft.Generator.CSharp.Input
 {
     /// <summary>
     /// Represents an input type to the generator.
     /// </summary>
-    public abstract class InputType
+    public abstract class InputType : InputDecoratedType
```

Comment:
this inheritance seems backwards to me.  InputType should be the base type shouldn't it?

Reply:
hello @m-nash we will need to emit decorator list for models, operations, parameters, clients.  So we define an top-level base type `InputDecoratedType` to contain the emitted decorator lists, the model/operation/client/parameter will inheritance this.  `InputType` is the base type for all data model, and `InputDecoratedType` is the base of all the tsp items which will contain the emitted decorator list.

Reply:
I think we do not need this layer of abstraction right now.
For instance, would we use `InputClient` and `InputModelType` in the same place by asserting they are both "things with decorator lists"?
We could simply add a property to those types without introducing this extra layer of type.

Reply:
Yes. After discuss, we think there is only one field to added, so do not need to create a new interface, we will add the field in every type which needed. 
I will remove this interface